### PR TITLE
setup: Upgrade to Python 3.11

### PR DESCRIPTION
**Pull request by achimnol**
- setup: Upgrade the main interpreter version to Python 3.11.2
- setup: Regenerate lockfiles for Python 3.11
- fix: Newly found type error

Other notable changes:
* `backend.ai-krunner-static-gnu` package changes the docker volume name from `backendai-krunner.v9.{ARCH}.static-gnu` to `backendai-krunner.v10.{ARCH}.static-gnu`.
* `backend.ai-krunner-alpine` package changes the docker volume name from `backendai-krunner.v7.{ARCH}.alpine3.{8,11}` to `backendai-krunner.v8.{ARCH}.musllinux_1_2`.

### CODE REVIEW

**Comment by achimnol** [comment]
TODO: update krunner-alpine!

**Comment by achimnol** [comment]
Now updated both:
https://github.com/lablup/backend.ai-krunner-static-gnu
https://github.com/lablup/backend.ai-krunner-alpine

**Comment by studioego** [comment]
I checked the branch and install Backend.AI.
Installing processing is successful.
However, When try to run agent,  I found agent error logs.
```console
2023-02-13 16:09:13.290 INFO ai.backend.agent.server [68900] Preparing kernel runner environments...
2023-02-13 16:09:13.309 DEBUG ai.backend.plugin.entrypoint [68900] scan_entrypoint_from_buildscript('backendai_krunner_v10'): Namespace path: /Users/dhsung/bai-dev/src/ai/backend
2023-02-13 16:09:13.322 DEBUG ai.backend.plugin.entrypoint [68900] scan_entrypoint_from_buildscript('backendai_krunner_v10'): current src: /Users/dhsung/bai-dev/src
2023-02-13 16:09:13.333 DEBUG ai.backend.plugin.entrypoint [68900] scan_entrypoint_from_plugin_checkouts('backendai_krunner_v10'): plugin parent dir: /Users/dhsung/bai-dev/plugins
2023-02-13 16:09:13.337 DEBUG ai.backend.agent.docker.kernel [68900] loading krunner pkg: ai.backend.krunner.alpine.plugin
2023-02-13 16:09:13.339 DEBUG ai.backend.agent.docker.kernel [68900] loading krunner pkg: ai.backend.krunner.static_gnu.plugin
2023-02-13 16:09:13.341 ERROR aiotools.server [68900] Worker 0: Error during context manager initialization
  + Exception Group Traceback (most recent call last):
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/server.py", line 325, in _wrapped_worker
  |     async with ctx:
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/server.py", line 119, in __aenter__
  |     return (await self._agen.__anext__())
  |     ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/server.py", line 680, in server_main_logwrapper
  |     async with server_main(loop, pidx, _args):
  |     ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/server.py", line 119, in __aenter__
  |     return (await self._agen.__anext__())
  |     ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/server.py", line 714, in server_main
  |     krunner_volumes = await kernel_mod.prepare_krunner_env(local_config)  # type: ignore
  |       ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/docker/kernel.py", line 489, in prepare_krunner_env
  |     async with TaskGroup() as tg:
  |       ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/taskgroup/base.py", line 40, in __aexit__
  |     raise TaskGroupError(eg.message, eg.exceptions) from None
  |     ^^^^^^^^^^^^^^^^^
  | aiotools.taskgroup.types.TaskGroupError: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/docker/kernel.py", line 371, in prepare_krunner_env_impl
    |     raise ValueError('Unrecognized "distro[version]" format string.')
    |     ^^^^^^^^^^^^^^^^^
    | ValueError: Unrecognized "distro[version]" format string.
    +------------------------------------
2023-02-13 16:09:13.279 INFO ai.backend.agent.server [68865] Backend.AI Agent 23.03.0dev0
2023-02-13 16:09:13.279 INFO ai.backend.agent.server [68865] runtime: Python 3.11.2 (env: /Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2)
2023-02-13 16:09:13.279 DEBUG ai.backend.agent.config [68865] debug mode enabled.
2023-02-13 16:09:13.282 INFO ai.backend.agent.server [68865] Using uvloop as the event loop backend
2023-02-13 16:09:13.345 INFO ai.backend.agent.server [68865] exit.
```
image

**Comment by studioego** [comment]
I check the issue on /src/ai.backend/agent/docker/kernel.py
https://github.com/lablup/backend.ai/blob/main/src/ai/backend/agent/docker/kernel.py#L371

When Backend.AI agent is running, following the parameter `distro` values are shown these values.
https://github.com/lablup/backend.ai/blob/main/src/ai/backend/agent/docker/kernel.py#L366

```console
distro: musl_1_2
distro: static-gnu
```

File: src/ai/backend/agent/docker/kernel.py [hunk 1]
```diff
@@ -363,19 +363,13 @@ async def get_repl_out_addr(self) -> str:
...
     docker = Docker()
     arch = get_arch_name()
     current_version = int(
         Path(
             pkg_resources.resource_filename(
-                f"ai.backend.krunner.{distro_name}", f"./krunner-version.{distro}.txt"
+                f"ai.backend.krunner.{entrypoint_name}", f"./krunner-version.{distro}.txt"
             )
         )
         .read_text()
```
Comment (studioego) [inline src/ai/backend/agent/docker/kernel.py:375]: I found error log when Backend.AI agent is running.

```console
2023-02-13 16:41:19.400 DEBUG ai.backend.agent.docker.kernel [81314] loading krunner pkg: ai.backend.krunner.static_gnu.plugin
2023-02-13 16:41:19.402 ERROR aiotools.server [81314] Worker 0: Error during context manager initialization
  + Exception Group Traceback (most recent call last):
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/server.py", line 325, in _wrapped_worker
  |     async with ctx:
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/server.py", line 119, in __aenter__
  |     return (await self._agen.__anext__())
  |     ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/server.py", line 680, in server_main_logwrapper
  |     async with server_main(loop, pidx, _args):
  |     ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/server.py", line 119, in __aenter__
  |     return (await self._agen.__anext__())
  |     ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/server.py", line 714, in server_main
  |     krunner_volumes = await kernel_mod.prepare_krunner_env(local_config)  # type: ignore
  |       ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/docker/kernel.py", line 483, in prepare_krunner_env
  |     async with TaskGroup() as tg:
  |       ^^^^^^^^^^^^^^^^^
  |   File "/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/aiotools/taskgroup/base.py", line 40, in __aexit__
  |     raise TaskGroupError(eg.message, eg.exceptions) from None
  |     ^^^^^^^^^^^^^^^^^
  | aiotools.taskgroup.types.TaskGroupError: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/Users/dhsung/bai-dev/src/ai/backend/agent/docker/kernel.py", line 375, in prepare_krunner_env_impl
    |     .read_text()
    |   File "/Users/dhsung/.pyenv/versions/3.11.2/lib/python3.11/pathlib.py", line 1058, in read_text
    |     with self.open(mode='r', encoding=encoding, errors=errors) as f:
    |     ^^^^^^^^^^^^^^^^^
    |   File "/Users/dhsung/.pyenv/versions/3.11.2/lib/python3.11/pathlib.py", line 1044, in open
    |     return io.open(self, mode, buffering, encoding, errors, newline)
    |     ^^^^^^^^^^^^^^^^^
    | FileNotFoundError: [Errno 2] No such file or directory: '/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/ai/backend/krunner/alpine/krunner-version.musl_1_2.txt'
    +------------------------------------
2023-02-13 16:41:19.407 INFO ai.backend.agent.server [81252] exit.
```
image

After I checked the log, I'm checking the  `/Users/dhsung/bai-dev/dist/export/python/virtualenvs/python-default/3.11.2/lib/python3.11/site-packages/ai/backend/krunner/alpine` directory.

```console
➜  alpine git:(setup/upgrade-to-py3.11) ✗ ls
__init__.py                              krunner-python.musllinux_1_2.txt         ttyd_linux.aarch64.bin
__pycache__                              krunner-version.musllinux_1_2.txt        ttyd_linux.x86_64.bin
krunner-env.musllinux_1_2.aarch64.tar.xz krunner-wheels.musllinux_1_2.dockerfile  versions.txt
krunner-env.musllinux_1_2.dockerfile     plugin.py
krunner-python.musllinux_1_2.dockerfile  requirements.txt
```

Error log shows that  ` FileNotFoundError: ***/krunner-version.musl_1_2.txt`. However, target directory's file name is `krunner-version.musllinux_1_2.txt`. 

So, I think, krunner file pattern is wrong.

**Review by studioego — CHANGES_REQUESTED** [review]

**Review by studioego — APPROVED** [review]
It works successfully,
image

However, I try to delete and reinstall Backend.AI.
When reinstalling after remove Backend.AI, I found pants error. 
https://github.com/lablup/backend.ai/issues/1046

Nevertheless, adding parameter `--pring-stacktrace`  to the file `install-dev.sh` and reinstalling Backend.AI  is installed successfully.

I don't understand these issue https://github.com/lablup/backend.ai/issues/1046

**Review by studioego — APPROVED** [review]
Also, on Linux x86_64 platform(Fedora Core 37), I delete default Backend.AI and reinstall Backend.AI this branch.
Installing processing is easily good.
It works successfully.
image

**Comment by studioego** [comment]
on Linux aarch platform(Ubunrtu 22.04), I delete default Backend.AI and reinstall Backend.AI this branch.
Installing processing is easily good.
It works successfully.
image

**Pull request merged**